[DOC-540] Update document  for window [master] (#534)
Macrometacorp/docs: docs/cep/query-guide/functions/unique/index.md

---
sidebar_position: 12
title: Unique (window)
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> This window is used with namespace i.e. `unique:windowName` This window retains and process unique events based on the given parameters.

Reviewer: ```suggestion
This window is used with a namespace such as `unique:windowName`. The window retains and processes unique events based on the given parameters.
```

| [length (Window)](unique/length)                    | A sliding length window that holds the latest unique events base in `unique.key`.                                                                                            |
| [lengthBatch (Window)](unique/lengthBatch)              | This is a batch (tumbling) window that holds `window.length` number of latest unique events.                                                                                 |
| [time (Window)](unique/time)                      | This is a sliding time window that holds the latest unique events that arrived during `window.time` .                                                                        |
| [timeBatch (Window)](unique/timeBatch)                 | This is a batch (tumbling) time window that is updated with the latest events based on `unique.key` tumbling time period `window.time` and time offset `start.time`.         |
| [timeLengthBatch (Window)](unique/timeLengthBatch) | The window tumbles upon the elapse of the time window `window.time`, or when a number of unique events have arrived i.e. `window.length`.                                    |

> ## Example

Reviewer: Please provide some context and explanation for the example. Pretend you are explaining it to me, a person who has never seen it before and is fuzzy on what windows and stream workers are. Use simple, plain language.
